### ISSUE TITLE
Slow startup with webengine (websettings)

### ISSUE THREAD

**Issue by rcorre** [qutebrowser/qutebrowser#2601]
qutebrowser takes ~5-10s to start up when I am running on my 'work' os. My 'personal' and 'work' OS run on the same PC, and are both similar* Archlinux installs, but there is a substantial difference in startup time ('personal' is ~1-2s). The slowness occurs only with `--backend webengine`, only occurs the first time I start it during a boot, and spends most of the time `initializing websettings`.

I'll try to dig more into this on my own, but wanted to post it here in case anyone had ideas or has seen something similar (also investigation is a bit slow when I have to reboot to reproduce it :roll_eyes:)

* Similar means same dotfiles and a lot of the same software (e.g. `dwm` as a window manager). It may be worth noting that 'personal' uses the standard kernel while 'work' uses the LTS kernel, for reasons relating to a whole fiasco of trying to manage two linux installs sharing the same EFI boot partition :angry:

Version info:
```
qutebrowser v0.10.1
Git commit: 
Backend: QtWebEngine

CPython: 3.6.1
Qt: 5.8.0
PyQt: 5.8.2

sip: 4.19.2
colorama: no
pypeg2: 2.15
jinja2: 2.9.6
pygments: 2.2.0
yaml: 3.12
cssutils: no
typing: yes
PyQt5.QtWebEngineWidgets: yes
pdf.js: no
Webkit: 538.1
SSL: OpenSSL 1.0.2k  26 Jan 2017

Style: QFusionStyle
Platform: Linux-4.9.24-1-lts-x86_64-with-arch, 64bit
Frozen: False
Imported from /usr/lib/python3.6/site-packages/qutebrowser
Qt library executable path: /usr/lib/qt/libexec, data path: /usr/share/qt
OS Version: 

--- /etc/os-release ---
NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
ID_LIKE=archlinux

Paths:
config: /home/rodencor/.config/qutebrowser
data: /home/rodencor/.local/share/qutebrowser
system_data: /usr/share/qutebrowser
cache: /home/rodencor/.cache/qutebrowser
download: /home/rodencor/Downloads
runtime: /run/user/1000/qutebrowser
```

**Comment by The-Compiler** [comment]
When initializing the web settings, afaik it's also the first time Qt creates a QtWebEngine "context" internally, i.e. does things like OpenGL init and stuff. I'm not sure there's anything we can do about that.

**Comment by lahwaacz** [comment]
I assume that the two systems share the same disk drive. What are the I/O levels during the startup? (you can use e.g. [iotop](https://www.archlinux.org/packages/community/any/iotop/)) Does [dropping cache](https://unix.stackexchange.com/questions/17936/setting-proc-sys-vm-drop-caches-to-clear-cache) instead of reboot make any difference?

**Comment by tuxlovesyou** [comment]
I noticed this as well when using one of my older dual i7 Nehelem laptops,
despite using a Samsung Evo 850 SSD.

On my beefy quad-core i7 Sandy Bridge laptop it is still slower than usual,
but not unreasonably slow.

**Comment by rcorre** [comment]
@lahwaacz they are on separate, but similar (Samsung Evo 850) SSDs. `iotop` only shows substantial disk activity after `initializing websettings`. Subsequent starts are fast even after dropping cache.

**Comment by The-Compiler** [comment]
@rcorre Did you even find out more here? I still doubt we can do anything here, as it's just QtWebEngine initializing things.

**Comment by rcorre** [comment]
Let me check when I start work today ... I feel like I haven't noticed this recently. It might have been a difference in lts/non-lts nvidia drivers and opengl init.

**Comment by The-Compiler** [comment]
@rcorre ping? :wink:

**Comment by rcorre** [comment]
Oops, sorry! Work OS is just as fast as the home one now, and I think it has been for a while.

**Comment by JC3** [comment]
I know this is closed and a bit old but it does come up on Google results.

One thing I found that helps a lot is, in the constructor of a window with QWebEngineViews, if you prime each view with blank HTML, it smooths out the UX significantly by shifting the web view initialization time to the window load time. E.g.:

    SomeMainWindow :: SomeMainWindow (...) {
        ...;
        ui->someWebView1->setHtml("<html></html>");
        ui->someWebView2->setHtml("<html></html>");
        ...;
    }

Also I've noticed that the view init times are significantly faster when done there compared to waiting until first use after the window is displayed.

Hopefully somebody finds that useful some day.

Using Qt 5.15.2 at the moment; Windows, MSVC 2019, release builds (PS debug build WebEngine init is *way* slower).
